### PR TITLE
Apply RDS API updates to CRD

### PR DESCRIPTION
**Pull request by jkatz**
Fixes aws-controllers-k8s/community#1311

Description of changes: This updates the API fields available to the RDS API.  This also adds support for creating Aurora Serverless v2 clusters.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by RedbackThomson — CHANGES_REQUESTED** [review]
Most of this looks like it can be merged as-is. Just commented on the performance insights addition to DBCluster - that will need to be fixed or we'll get a repeat of a previous bug.

File: apis/v1alpha1/db_cluster.go [hunk 1]
```diff
@@ -131,28 +213,53 @@ type DBClusterSpec struct {
...
+	// Valid for: Aurora DB clusters only
 	EnableIAMDatabaseAuthentication *bool `json:"enableIAMDatabaseAuthentication,omitempty"`
+	// A value that indicates whether to turn on Performance Insights for the DB
+	// cluster.
+	//
+	// For more information, see Using Amazon Performance Insights (https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PerfInsights.html)
+	// in the Amazon RDS User Guide.
+	//
+	// Valid for: Multi-AZ DB clusters only
+	EnablePerformanceInsights *bool `json:"enablePerformanceInsights,omitempty"`
```
[the remark this answers is not in the document]
Comment (jkatz) [reply apis/v1alpha1/db_cluster.go:247]: @RedbackThomson :+1: -- this seems a bit brittle though, is there a way we can prevent the issue #77 fixed from being reintroduced?

[the remark this answers is not in the document]
Comment (jkatz) [reply apis/v1alpha1/db_cluster.go]: @RedbackThomson OK, I believe I got the changes from #77 in.

**Review by jaypipes — APPROVED** [review]
Awesome, thanks @jkatz! :)

**Comment by jkatz** [comment]
@RedbackThomson I've adjusted the patch so the renames for `EnablePerformanceInsights` in `DBCluster` mirror the work in `DBInstance`.

That said, I saw aws-controllers-k8s/runtime#92 and wanted to check if the renaming was still necessary. There is the open question if we want to make the boolean names be declarative, but from a technical perspective, do we need to rename thes values still?

**Pull request merged**